### PR TITLE
Check duplicate field when create the index

### PR DESCRIPTION
**Pull request by darionyaphet**

### CODE REVIEW

File: src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp [hunk 1]
```diff
@@ -16,12 +16,20 @@ void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
     auto &edgeName = req.get_edge_name();
     auto &fieldNames = req.get_fields();
     if (fieldNames.empty()) {
-        LOG(ERROR) << "Edge's Field should not empty";
+        LOG(ERROR) << "Edge Index's field should not empty";
```
Comment (Amber1990Zhang) [inline src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp]: The index field of an edge type should not be empty.

File: src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp [hunk 2]
```diff
@@ -16,12 +16,20 @@ void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
...
         handleErrorCode(cpp2::ErrorCode::E_INVALID_PARM);
         onFinished();
         return;
     }
 
+    std::set<std::string> columnSet(fieldNames.begin(), fieldNames.end());
+    if (fieldNames.size() != columnSet.size()) {
+        LOG(ERROR) << "Edge Index have the conflict field";
```
Comment (Amber1990Zhang) [inline src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp]: Conflict field in the edge index.

File: src/meta/processors/indexMan/CreateTagIndexProcessor.cpp [hunk 3]
```diff
@@ -16,11 +16,18 @@ void CreateTagIndexProcessor::process(const cpp2::CreateTagIndexReq& req) {
     auto &tagName = req.get_tag_name();
     auto &fieldNames = req.get_fields();
     if (fieldNames.empty()) {
-        LOG(ERROR) << "Tag's Field should not empty";
+        LOG(ERROR) << "Tag Index's field should not empty";
```
Comment (Amber1990Zhang) [inline src/meta/processors/indexMan/CreateTagIndexProcessor.cpp]: ditto

File: src/meta/processors/indexMan/CreateTagIndexProcessor.cpp [hunk 4]
```diff
@@ -16,11 +16,18 @@ void CreateTagIndexProcessor::process(const cpp2::CreateTagIndexReq& req) {
...
         handleErrorCode(cpp2::ErrorCode::E_INVALID_PARM);
         onFinished();
         return;
     }
+    std::set<std::string> columnSet(fieldNames.begin(), fieldNames.end());
+    if (fieldNames.size() != columnSet.size()) {
+        LOG(ERROR) << "Tag Index have the conflict field";
```
Comment (Amber1990Zhang) [inline src/meta/processors/indexMan/CreateTagIndexProcessor.cpp]: ditto

**Comment by bright-starry-sky** [comment]
Well done! waiting inline comments.

File: src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp (hunk 1 above)
Comment (darionyaphet) [reply src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp]: Good Point ~~

**Pull request merged**